File: aoc_6.jl
function simulate_fish(fish, days)
    counter = [count(==(n), fish) for n in 1:9]
    for _ in 1:days
        n = counter[1]
        counter = circshift(counter, -1)
        counter[7] += n
    end
    sum(counter)
end

fish = 1 .+ parse.(Int, split(readline("data/aoc_6"), ","))
println(simulate_fish(fish, 80), "\n", simulate_fish(fish, 256))

File: aoc_6.R
simulate_fish <- function(fish, days) {
  counter <- as.numeric(tabulate(fish, 9))
  for (i in 1:days) {
    n <- counter[1]
    counter <- c(counter[-1], counter[1])
    if (n) counter[7] <- counter[7] + n
  }
  sum(counter)
}

fish <- 1 + scan("data/aoc_6", sep = ",")
c(part1 = simulate_fish(fish, 80),
  part2 = simulate_fish(fish, 256)) |> print(digits = 18)
